### PR TITLE
Document OpenTelemetry drop-in support

### PR DESCRIPTION
**Pull request by mcculls**
### What does this PR do? What is the motivation?

Various language SDKs support adding OpenTelemetry instrumentations alongside their built-in instrumentations.
This PR documents how to achieve that - initially for Java, but other languages will add their steps to this PR.

### Merge instructions

- [x] Please merge after reviewing
- [ ] Add instructions for other languages before merging

### CODE REVIEW

File: content/en/opentelemetry/interoperability/drop_in_support.md
Comment (bm1549) [inline content/en/opentelemetry/interoperability/drop_in_support.md:16]:
```suggestion
Instrumentation is the act of adding observability code to an application. Automatic instrumentation is a way to instrument applications and libraries without directly modifying their source code. Both OpenTelemetry and Datadog provide instrumentations as part of their SDKs, with varying amounts of coverage between them.
```

Comment (bm1549) [reply content/en/opentelemetry/interoperability/drop_in_support.md:16]: Can provide links for 
Instrumentation: https://docs.datadoghq.com/tracing/trace_collection/

Automatic instrumentation: https://docs.datadoghq.com/tracing/trace_collection/automatic_instrumentation/?tab=singlestepinstrumentationbeta

Comment (bm1549) [reply content/en/opentelemetry/interoperability/drop_in_support.md:16]: Aside, it'd be good to align definitions between those pages and our glossary

https://docs.datadoghq.com/tracing/glossary/

File: content/en/opentelemetry/interoperability/drop_in_support.md
Comment (bm1549) [inline content/en/opentelemetry/interoperability/drop_in_support.md:20]: Let's move away from language pages and lean into language tabs, similar to what we have for [context propagation](https://github.com/DataDog/documentation/pull/23726)

File: content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md
Comment (bm1549) [inline content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md:14]: This can be common to all languages

File: content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md
Comment (bm1549) [inline content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md:18]: This can be a table with all languages

File: content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md
Comment (bm1549) [inline content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md:52]: I'm a bit torn between having this here (one view for all languages) and having it on the existing [language compatibility page](https://docs.datadoghq.com/tracing/trace_collection/compatibility/java?tab=graalvm#networking-framework-compatibility) like we do for Camel

What do you think about having a section on the language-specific compatibility pages and a link from here to those pages?

File: content/en/opentelemetry/interoperability/_index.md
Comment (zacharycmontoya) [inline content/en/opentelemetry/interoperability/_index.md:21]: What if we rename the shortname `drop_in_support` and the long name `Using OpenTelemetry Instrumentations` to `OpenTelemetry Instrumentation Support`. Thoughts @mcculls @tushar-datadog ?

File: content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md
Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md:52]: +1 to linking from this OpenTelemetry page to the corresponding language-specific automatic instrumentation page

File: content/en/opentelemetry/interoperability/_index.md
Comment (zacharycmontoya) [reply content/en/opentelemetry/interoperability/_index.md:21]: The terms are a little hand-wavey, but it seems like the term may be `Instrumentation Library` for the most up-to-date OpenTelemetry docs: https://opentelemetry.io/docs/specs/otel/overview/#instrumentation-libraries

File: content/en/opentelemetry/interoperability/drop_in_support.md
Comment (zacharycmontoya) [reply content/en/opentelemetry/interoperability/drop_in_support.md:16]: I think the description here does a good job of describing instrumentation libraries, which could be a good basis for our description here: https://opentelemetry.io/docs/concepts/instrumentation/libraries/

File: content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md
Comment (wantsui) [inline content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md:30]: What steps are needed to do this?

File: content/en/opentelemetry/interoperability/drop_in_support.md
Comment (zacharycmontoya) [reply content/en/opentelemetry/interoperability/drop_in_support.md:16]: I added a meta-comment to Ayan's PR and I think it may also apply here https://github.com/DataDog/documentation/pull/23787/files#r1664958998

File: content/en/opentelemetry/interoperability/_index.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/_index.md:20]: `Instrumentation Library` is only part of the story for Java - that defines library code that interacts with the OTel APIs (whether that is code shipped with the library, or available separately using library hooks to call into OTel.)

Most Java instrumentation is what they call [zero-code](https://opentelemetry.io/docs/concepts/instrumentation/zero-code/) or `agent-based` which involves bytecode manipulation.

Both `agent-based` instrumentation and `Instrumentation Libraries` fall under `automatic instrumentation`.

Comment (mcculls) [reply content/en/opentelemetry/interoperability/_index.md:20]: Going to go with `Instrumentation Libraries` because it's less vague than the other options - it's not quite the full story for Java, but we can cover that in the Java specific text

File: content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md
Comment (mcculls) [reply content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md]: Adding the necessary config detail to the language-specific automatic instrumentation page introduced too much complexity, so what I've done is list the additional Java frameworks that are now covered by adding OpenTelemetry instrumentations there with a link back to here where I can cover how to add them.

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (bm1549) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:33]: A bit nit picky, but can we put a placeholder here like `TBA`?

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (bm1549) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:52]: Since this is common across all languages, should we have it in an above "Prerequisites" section, potentially including "Language support"?

File: content/en/tracing/trace_collection/compatibility/java.md
Comment (bm1549) [inline content/en/tracing/trace_collection/compatibility/java.md:413]: Can we have this use the Java tab by default?

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (bm1549) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:63]: This is common across languages and can be at the top

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (bm1549) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:85]: Are these going to be filled in as part of this PR? If not, consider removing these or commenting them out for now

**Review by bm1549 — COMMENTED** [review]
Love the single page here! It makes it so much easier for our customers to see what they need to do at a glance

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mabdinur) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:29]:
```suggestion
| Python   |  v2.10.0         |
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mabdinur) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:30]:
```suggestion
| Ruby     |    v2.1.0                   |
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mabdinur) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:21]: Currently if a user runs their applications with automatic instrumentation (ex ddtrace-run), the tracer will auto-enable all/most ddtrace integrations. We should provide customer's with steps/examples on how to replace ddtrace integrations with an equivalent opentelemetry instrumentation.

Side note: Most client libraries do not have a single configuration that can disable ALL ddtrace instrumentations. User's must disable ddtrace integrations individually. This user experience seems less than ideal.

Comment (mabdinur) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:21]: cc: @tushar-datadog

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:63]: System properties are specific to Java - so this can't be moved up

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:85]: These are common placeholders for the different language experts to fill in - if they decide a section isn't required/applicable to their language then it could be removed, but I would like to keep some consistency across the tags where possible.

File: content/en/tracing/trace_collection/compatibility/java.md
Comment (mcculls) [reply content/en/tracing/trace_collection/compatibility/java.md:413]: Yes, I have this locally but I haven't pushed the latest batch of edits

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:21]: Yes, we should warn about the possibility of duplicate spans here.

But how to resolve that will vary across each language - this is one area of configuration which varies widely from language to language (even in OTel the way you enable/disable Java instrumentations is very specific to the Java tracer.)

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: We can - but I'd hoped to have this all filled in before this was published

If we want to publish this for selected languages first then I can add `TBA` here and comment out their tabs...

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:85]: I've commented out the empty tabs for now and put `TBA` placeholders in the table - that way we can merge this once we're happy with the general and Java specific content (each language can then work on their own PR)

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: I've added TBA for the missing versions

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (bm1549) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:52]: I think that's fine here. Main goal is for this to be clear to users at a glance

**Review by bm1549 — APPROVED** [review]
Great work Stuart!

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:21]: Added a general warning to the preamble

File: content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md
Comment (mcculls) [reply content/en/tracing/trace_collection/automatic_instrumentation/otel/java.md]: I've reworked the earlier text to refer to OpenTelemetry's example build as an example

**Comment by buraizu** [comment]
Thanks for the PR, I've created DOCS-8443 for documentation team review

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mtoffl01) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]: We may want to restructure where this statement lives because its not a prereq for all languages (e.g, Go)

Comment (bm1549) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]: @mtoffl01 How would you suggest restructuring this?

This was called out [here](https://github.com/DataDog/documentation/pull/23819#discussion_r1672500748), but we couldn't identify a better way to frame the information

Comment (mtoffl01) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]: One option would be to leave it where it is and add a qualifier with the languages that require it (e.g, "If you use Java, X, Y or Z, then you must set the `DD_TRACE_OTEL_ENABLED` environment variable to `true`.")
Alternatively we can cut this section and just list it under the language tabs that require this env var - I see the env var is mentioned in the Java specific section under "Getting Started" but I if the point is to highlight it **more**, then we could move it out to its own section within the Language specific instructions, like before "compatibility requirements" or something.

Comment (mtoffl01) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]: Also: IMO there is some secondary "harm" in documenting it as a prereq for all languages when it comes to troubleshooting/support. 
Our customers and our support team will use our docs as a source of truth and may incorrectly attribute unexpected behavior to the env var (or lack thereof)

Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]: The general view was that it was better to have this called out here for visibility, even though it is not strictly necessary for a couple of languages: https://github.com/DataDog/documentation/pull/23819#discussion_r1673680945 (setting it on those languages won't do any harm)

Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]: Adding qualifiers, etc. makes it more likely customers will miss out this detail or get confused about when to set it.

Either we just remove it and have each language decide whether to call out that you need to set it (i.e. how it was before) or we leave it in place as a blanket setting.

IMHO removing it is more likely to lead to support tickets than leaving it in place...

Comment (mtoffl01) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]: Discussed with Stuart on slack — we decided to keep the section, but add a qualifier that this is not required in Go.

Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]:
```suggestion
Before adding OpenTelemetry instrumentation libraries, set the `DD_TRACE_OTEL_ENABLED` environment variable to `true`.
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:43]:
```suggestion
| Go       | Not supported                      |
| Node.js  | Not supported                      |
| PHP      | Not supported                      |
| .NET     | Not supported                      |

```
Small tweak, but I'd suggest using "Not supported", "N/A", or " - " instead of TBA. This ensures we're documenting the current state of support rather than implying future functionality.

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:67]:
```suggestion
1. Set the `dd.trace.otel.enabled` system property or the `DD_TRACE_OTEL_ENABLED` environment variable to `true`.
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:68]:
```suggestion
2. Copy the OpenTelemetry extension jar containing the instrumentation to the same container as the application.
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:70]:
```suggestion
3. Set the `otel.javaagent.extensions` system property or the `OTEL_JAVAAGENT_EXTENSIONS` environment variable to the path to the extension jar.
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:90]:
```suggestion
## Verified OpenTelemetry extensions
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:86]:
```suggestion
A comma-separated list of paths to extension jar files or folders containing extension jar files.
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:88]:
```suggestion
OpenTelemetry's [Agent Configuration](https://opentelemetry.io/docs/zero-code/java/agent/configuration/) page describes additional properties that are also recognized by the Datadog SDK.
```

**Review by brett0000FF — CHANGES_REQUESTED** [review]
Looks great, thanks for adding the content! I made a few smalls suggestions. The only blocking suggestion is we need to update the Markdown links to reference style instead of inline. 🚀

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:43]: On reflection `Not supported` is not quite right because the functionality is actually released, but has not yet been announced. We expect these rows to be filled in quickly once this initial PR is merged, so I think `-` makes more sense.

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: Done, thanks: 77158445e1

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: Ah, good thinking. Makes sense to me. 👍

**Review by brett0000FF — APPROVED** [review]
🚀

**Comment by mcculls** [comment]
Thanks for the review @brett0000FF - is there anything else I need to do before this gets merged?

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (tushar-datadog) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:54]: provides

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (tushar-datadog) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:39]: Is there a reason we are not calling out min languages for these four?

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:54]: There are also 2 uses of 'provides' in this sentence which is not great - I'll fix that

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:39]: Munir helped fill in some of the versions, the rest were going to be filled in as their tabs were added (the idea was to merge this first, to make it easier for each language to work on a separate follow-up PR rather than have this PR as a bottleneck)

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: c0487e49dc

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: d5c6fb8e10

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (zacharycmontoya) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:46]: I'm adding a separate PR for more details on .NET, but here's the version number for immediate plugging in

```suggestion
| .NET     | v2.53.0                  |
```

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mtoffl01) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:43]:
```suggestion
| Go       |  v1.65.0 (Span event support planned for v1.67.0) |
```

Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md:43]: Thanks - I think `v1.65.0` is enough detail for now (as plans always change)

Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: bfb59548b2

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (link04) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:45]: For PHP it `DD_TRACE_OTEL_ENABLED ` was added in [0.94.0](https://github.com/DataDog/dd-trace-php/releases/tag/0.94.0), lets please use that version to satisfy products expectations.

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (mcculls) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: OK, version table is now complete - thanks to everyone who contributed!

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**
